File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconRequestHandler.java
package org.yuzu.yuzu_emu.utils;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import com.squareup.picasso.Picasso;
import com.squareup.picasso.Request;
import com.squareup.picasso.RequestHandler;

import org.yuzu.yuzu_emu.NativeLibrary;

import java.nio.IntBuffer;

public class GameIconRequestHandler extends RequestHandler {
    @Override
    public boolean canHandleRequest(Request data) {
        return "content".equals(data.uri.getScheme());
    }

    @Override
    public Result load(Request request, int networkPolicy) {
        String gamePath = request.uri.toString();
        byte[] data = NativeLibrary.GetIcon(gamePath);
        BitmapFactory.Options options = new BitmapFactory.Options();
        options.inMutable = true;
        Bitmap bitmap = BitmapFactory.decodeByteArray(data, 0, data.length, options);
        return new Result(bitmap, Picasso.LoadedFrom.DISK);
    }
}

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconRequestHandler.kt
// SPDX-FileCopyrightText: 2023 yuzu Emulator Project
// SPDX-License-Identifier: GPL-2.0-or-later

package org.yuzu.yuzu_emu.utils

import android.graphics.BitmapFactory
import com.squareup.picasso.Picasso
import com.squareup.picasso.Request
import com.squareup.picasso.RequestHandler
import org.yuzu.yuzu_emu.NativeLibrary

class GameIconRequestHandler : RequestHandler() {
    override fun canHandleRequest(data: Request): Boolean {
        return "content" == data.uri.scheme
    }

    override fun load(request: Request, networkPolicy: Int): Result {
        val gamePath = request.uri.toString()
        val data = NativeLibrary.GetIcon(gamePath)
        val options = BitmapFactory.Options()
        options.inMutable = true
        val bitmap = BitmapFactory.decodeByteArray(data, 0, data.size, options)
        return Result(bitmap, Picasso.LoadedFrom.DISK)
    }
}
